File: JOCL/ArraySumBenchmark/src/GPUBenchmarkData.java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.PrintStream;


public class GPUBenchmarkData {
	
	
	public static void main(String args[]) throws FileNotFoundException{
		
<<<<<<< HEAD
		//BenchmarkGPULoop data = new BenchmarkGPULoop();
		//BenchmarkCPULoop datacpu = new BenchmarkCPULoop();
		BenchmarkOCLCPU dataoclcpu = new BenchmarkOCLCPU();


		int minArraySize = 500;
		int maxArraySize = 10000;
		int minIters = 0;
		int maxIters = 1000;
=======
		BenchmarkGPULoop data = new BenchmarkGPULoop();
		//BenchmarkCPULoop datacpu = new BenchmarkCPULoop();
		//BenchmarkOCLCPU dataoclcpu = new BenchmarkOCLCPU();
		
		int minArraySize = 500;
		int maxArraySize = 10000;
		int minIters = 1;
		int maxIters = 100000;
>>>>>>> 17f84df76dd0d1428c04a63b074859006c5ff240
		int k=0;
		int z=0;
		int sizeStep = 100;
		int iterStep = 10;
		int row = (maxArraySize - minArraySize)/sizeStep;
		float[][] flopValues = new float[row][8];
		PrintStream output = new PrintStream(new File("data.txt"));
		
		for(int i = minArraySize; i <= maxArraySize; i+=100){
			k++;
			z = 0;
			System.out.println(i);
			output.println();
<<<<<<< HEAD
			for(int j = minIters; j<=maxIters; j+=100){
				//flopValues[k][z] = data.runBenchmark(j, i, i+1);
				//flopValues[k][z] = datacpu.runBenchmarkCPU(j, i, i+1);
				flopValues[k][z] = dataoclcpu.runBenchmarkOCLCPU(j, i, i+1);
=======
			for(int j = minIters; j<=maxIters; j*=10){
				flopValues[k][z] = data.runBenchmark(j, i, i+1);
				//flopValues[k][z] = datacpu.runBenchmarkCPU(j, i, i+1);
				//flopValues[k][z] = dataoclcpu.runBenchmarkOCLCPU(j, i, i+1);
>>>>>>> 17f84df76dd0d1428c04a63b074859006c5ff240
				output.print(flopValues[k][z] + " ");
				z++;
				System.out.println(z);
				
			}
		}		
	

	
	}

}

File: OpenCL/MacOSX/ArraySumBenchmark/GPUBenchmarkData.cpp
<<<<<<< HEAD


=======
//
//  GPUBenchmarkData.cpp
//  ArraySumBenchmark
//
//  Created by Joseph Jennings on 11/23/13.
//  Copyright (c) 2013 Joseph Jennings. All rights reserved.
//

#include <iostream>
#include "GPU.h"
#include "ArraySumUtil.h"
#include "Stopwatch.h"
#include <OpenCL/OpenCL.h>
#include "BenchmarkGPU.h"
#include "BenchmarkCPU.h"
#include "BenchmarkOCLCPU.h"


int main (int argc, const char * argv[]){
    
    BenchmarkGPU datagpu;
    BenchmarkCPU datacpu;
    BenchmarkOCLCPU dataoclcpu;
    
    int minIters = 1;
    int maxIters = 4096;
    int minSize = 500;
    int maxSize = 2500;
    int a = 0;
    int b = 0;
    
    float** data = new float*[maxSize/minSize];
    for (int i = 0; i < 11; ++i)
        data[i] = new float[11];
    
    for(int i = minSize; i < maxSize; i+=100){
        std::cout << "Array Size: " << i << std::endl;
        for(int j = minIters; j < maxIters; j*=2){
            std::cout << "Iterations: " << j << std::endl;
            data[a][b] = datagpu.runGPUBenchmark(j, i, i+1);
            //data[a][b] = datacpu.runCPUBenchmark(j, i, i+1);
            //data[a][b] = dataoclcpu.runBenchmarkOCLCPU(j, i, i+1);
            b++;
        }
        a++;
        std::cout << std::endl;
    }
    
    return 0;
    
    
}
>>>>>>> 17f84df76dd0d1428c04a63b074859006c5ff240
